browser/src/control/Control.JSDialog.js

```diff
@@ -129,350 +133,155 @@ L.Control.JSDialog = L.Control.extend({
 		return null;
 	},
 
-	onJSDialog: function(e) {
-		var that = this;
-		var posX = 0;
-		var posY = 0;
-		var data = e.data;
-		var callback = e.callback;
-		var isSnackbar = data.type === 'snackbar';
-		var isModalPopup = data.type === 'modalpopup' || isSnackbar;
-		var canHaveFocus = !isSnackbar && data.id !== 'busypopup' && !data.isMention;
-		var focusWidgetId = data.init_focus_id;
-		var isOnlyChild = false;
-
-		if (data.action === 'fadeout')
-		{
-			if (data.id && this.dialogs[data.id]) {
-				var container = this.dialogs[data.id].container;
-				L.DomUtil.addClass(container, 'fadeout');
-				container.onanimationend = function() { that.close(data.id); };
-				// be sure it will be removed
-				setTimeout(function() { that.close(data.id); }, 700);
-			}
-			return;
-		}
-		else if (data.action === 'close')
-		{
-			this.close(data.id, false);
-
-			// Manage focus
-			var dialogs = Object.keys(this.dialogs);
-			if (dialogs.length) {
-				var lastKey = dialogs[dialogs.length - 1];
-				var container = this.dialogs[lastKey].container;
-				container.focus();
-				var initialFocusElement =
-					container.querySelector('[tabIndex="0"]:not(.jsdialog-begin-marker)');
-				initialFocusElement.focus();
-			}
-			else if (!this.hasDialogOpened()) {
-				this._map.fire('editorgotfocus');
-			}
-
-			return;
+	fadeOutDialog: function(instance) {
+		if (instance.id && this.dialogs[instance.id]) {
+			var container = this.dialogs[instance.id].container;
+			L.DomUtil.addClass(container, 'fadeout');
+			container.onanimationend = function() { instance.that.close(instance.id); };
+			// be sure it will be removed
+			setTimeout(function() { instance.that.close(instance.id); }, 700);
 		}
+	},
 
-		var toRemove = null;
-		if (this.dialogs[data.id]) {
-			posX = this.dialogs[data.id].startX;
-			posY = this.dialogs[data.id].startY;
-			toRemove = this.dialogs[data.id].container;
+	getOrCreateOverlay: function(instance) {
+		// Dialogue overlay which will allow automatic positioning and cancellation of the dialogue if cancellable.
+		var overlay = L.DomUtil.get(instance.id + '-overlay');
+		if (!overlay) {
+			overlay = L.DomUtil.create('div', 'jsdialog-overlay ' + (instance.cancellable ? 'cancellable' : ''), instance.containerParent);
```

Comment:
see removed line 295:
there it was: `'(data.cancellable && !data.hasOverlay ? 'cancellable' : '')`
I recently added hasOverlay to the condition so it will be blue overlay for modals